test/components/tooltip.spec.tsx

```diff
@@ -25,107 +38,121 @@ class SampleDriver extends DriverBase {
+                expect(anchorBounds.left).to.equal(tooltipBounds.left);
+            });
+        }
+        if (expectations.aligmnentRight) {
+            it('should be alignet to right', () => {
+                const tooltipBounds = driver.tooltipBounds;
+                const anchorBounds = driver.anchorBounds;
+                expect(anchorBounds.left + anchorBounds.width).to.equal(tooltipBounds.left + tooltipBounds.width);
+            });
+        }
+        if (expectations.aligmnentTop) {
+            it('should be alignet to top', () => {
+                const tooltipBounds = driver.tooltipBounds;
+                const anchorBounds = driver.anchorBounds;
+                expect(anchorBounds.top).to.equal(tooltipBounds.top);
+            });
+        }
+        if (expectations.aligmnentBottom) {
+            it('should be alignet to bottom', () => {
+                const tooltipBounds = driver.tooltipBounds;
+                const anchorBounds = driver.anchorBounds;
+                expect(anchorBounds.top + anchorBounds.height).to.equal(tooltipBounds.top + tooltipBounds.height);
+            });
+        }
     });
+}
 
-    describe('render with left position', () => {
-        let driver: any;
-        beforeEach(() => {
-            driver = renderWithProps(clientRenderer, {position: 'left'});
-        });
-
-        it('should be alignet to left', () => {
-            const tooltipBounds = driver.tooltipBounds;
-            const anchorBounds = driver.anchorBounds;
-            expect(anchorBounds.left).to.equal(tooltipBounds.left + tooltipBounds.width - driver.tooltipMargins.left);
-        });
-
-        it('should be centerd verticaly', () => {
-            const tooltipBounds = driver.tooltipBounds;
-            const anchorBounds = driver.anchorBounds;
-            expect(anchorBounds.top + anchorBounds.height / 2).to.equal(tooltipBounds.top + tooltipBounds.height / 2);
-        });
-    });
-
-    describe('render with right position', () => {
-        let driver: any;
-        beforeEach(() => {
-            driver = renderWithProps(clientRenderer, {position: 'right'});
-        });
-
-        it('should be alignet to left', () => {
-            const tooltipBounds = driver.tooltipBounds;
-            const anchorBounds = driver.anchorBounds;
-            expect(anchorBounds.left + anchorBounds.width).to.equal(tooltipBounds.left - driver.tooltipMargins.left);
-        });
+describe('<Tooltip/>', () => {
+    const clientRenderer = new ClientRenderer();
+    afterEach(() => clientRenderer.cleanup());
 
-        it('should be centerd verticaly', () => {
-            const tooltipBounds = driver.tooltipBounds;
-            const anchorBounds = driver.anchorBounds;
-            expect(anchorBounds.top + anchorBounds.height / 2).to.equal(tooltipBounds.top + tooltipBounds.height / 2);
-        });
-    });
+    testPosition(clientRenderer, 'top', {positionTop: true, centeredHorizontaly: true});
```

Comment:
`clientRenderer` maybe move it to default function param?

Reply:
i reused the same `clientRenderer` just to avoid manual `.cleanup()` call, but i could be defined inside `testPosition` without any problem 👍